azure-iot-device/azure/iot/device/provisioning/abstract_provisioning_device_client.py

```diff
@@ -32,30 +32,75 @@ def __init__(self, provisioning_pipeline):
         self._provisioning_pipeline = provisioning_pipeline
 
     @classmethod
-    def create_from_security_client(cls, security_client, protocol_name):
+    def create_from_symmetric_key(
+        cls, provisioning_host, registration_id, id_scope, symmetric_key, protocol_choice=None
+    ):
         """
-        Creates different types of provisioning clients which can enable devices to communicate with Device Provisioning
-        Service based on parameters passed.
-        :param security_client: Instance of Security client object which can be either of SymmetricKeySecurityClient,  TPMSecurtiyClient or X509SecurityClient
-        :param protocol_name: A string representing the name of the communication protocol the user wants
-        :return: A specific provisioning client based on parameters and validations.
+        Create a client which can be used to run the registration of a device with provisioning service
+        using Symmetric Key authentication.
+        :param provisioning_host: Host running the Device Provisioning Service. Can be found in the Azure portal in the
+        Overview tab as the string Global device endpoint
+        :param registration_id: The registration ID is used to uniquely identify a device in the Device Provisioning Service.
+        The registration ID is alphanumeric, lowercase string and may contain hyphens.
+        :param id_scope: The ID scope is used to uniquely identify the specific provisioning service the device will
+        register through. The ID scope is assigned to a Device Provisioning Service when it is created by the user and
+        is generated by the service and is immutable, guaranteeing uniqueness.
+        :param symmetric_key: The key which will be used to create the shared access signature token to authenticate
+        the device with the Device Provisioning Service. By default, the Device Provisioning Service creates
+        new symmetric keys with a default length of 32 bytes when new enrollments are saved with the Auto-generate keys
+        option enabled. Users can provide their own symmetric keys for enrollments by disabling this option within
+        16 bytes and 64 bytes and in valid Base64 format.
+        :param protocol_choice: The choice for the protocol to be used. This is optional and will default to protocol MQTT currently.
+        :return: A ProvisioningDeviceClient which can register via Symmetric Key.
         """
-        protocol_name = protocol_name.lower()
+        if protocol_choice is not None:
```

Comment:
> is not None [](start = 26, length = 12)

Why are you specifically comparing to None?  This implies that you want to accept other falsy values that aren't None, like "", {}. [], 0, and False.  I don't think this is what you want.

Reply:
I will flip it. I think I wanted to do if its None default to mqtt.

---
In reply to: [294435666](https://github.com/Azure/azure-iot-sdk-python-preview/pull/109#discussion_r294435666) [](ancestors = 294435666)